google/cloud/bigtable/testing/table_integration_test.cc

```diff
@@ -30,6 +31,57 @@ std::string TableTestEnvironment::instance_id_;
 std::string TableTestEnvironment::cluster_id_;
 std::string TableTestEnvironment::zone_;
 std::string TableTestEnvironment::replication_zone_;
+google::cloud::internal::DefaultPRNG TableTestEnvironment::generator_;
+std::string TableTestEnvironment::table_id_;
+
+void TableTestEnvironment::SetUp() {
+  generator_ = google::cloud::internal::MakeDefaultPRNG();
+
+  auto admin_client = bigtable::CreateDefaultAdminClient(
+      TableTestEnvironment::project_id(), ClientOptions());
+  auto table_admin =
+      bigtable::TableAdmin(admin_client, TableTestEnvironment::instance_id());
+
+  std::string const family1 = "family1";
+  std::string const family2 = "family2";
+  std::string const family3 = "family3";
+  std::string const family4 = "family4";
+  bigtable::TableConfig table_config =
+      bigtable::TableConfig({{family1, bigtable::GcRule::MaxNumVersions(10)},
+                             {family2, bigtable::GcRule::MaxNumVersions(10)},
+                             {family3, bigtable::GcRule::MaxNumVersions(10)},
+                             {family4, bigtable::GcRule::MaxNumVersions(10)}},
+                            {});
+
+  table_id_ = RandomTableId();
+  ASSERT_STATUS_OK(table_admin.CreateTable(table_id_, table_config));
+}
+
+void TableTestEnvironment::TearDown() {
+  auto admin_client = bigtable::CreateDefaultAdminClient(
+      TableTestEnvironment::project_id(), ClientOptions());
+  auto table_admin =
+      bigtable::TableAdmin(admin_client, TableTestEnvironment::instance_id());
+
+  ASSERT_STATUS_OK(table_admin.DeleteTable(table_id_));
+}
+
+std::string TableTestEnvironment::CreateRandomId(std::string const& prefix,
+                                                 int count) {
+  return prefix +
+         google::cloud::internal::Sample(
+             generator_, count, "abcdefghijklmnopqrstuvwxyz0123456789");
+}
+
+std::string TableTestEnvironment::RandomTableId() {
+  constexpr int RANDOM_CHARACTERS = 16;
```

Comment:
What is our naming convention for constants?

Reply:
A mess. In principle should be the same as the Google Style Guide.  But we made a mess of them a while back, and the clang-tidy rules need fixing.  I would prefer to tackle that separately these were the constant names I refactored from specific tests.

Reply:
I fixed these to be `kCamelCase`. Apparently our clang-tidy rules are so messed up that anything is accepted here.